src/Components/AdvancedSearch/Internal/QueryBuilder/QueryBuilder.tsx

```diff
@@ -0,0 +1,203 @@
+        }
+        setIsSearchDisabled(!checkIsValidQuery());
+    }, []);
+
+    const updateQuerySnippet = (rowId: string, rowValue: QueryRowType) => {
+        querySnippets.current.set(rowId, rowValue);
+    };
+
+    const onChangeProperty = useCallback(
+        (
+            rowId: string,
+            propertyName: string,
+            propertyType: PropertyValueType
+        ) => {
+            // Either create new property entry or modify existing one
+            propertyNames.current.set(rowId, propertyName);
+            const propertyNameArray = propertyNames.current.values();
+            // Then send to parent component
+            updateColumns(new Set<string>(propertyNameArray));
+            // Reset validity and update snippet, string is the only field with no default value and blank is not a valid input
+            if (propertyType !== 'string') {
+                validityMap.current.set(rowId, true);
+            } else {
+                validityMap.current.set(rowId, false);
+            }
+            setIsSearchDisabled(!checkIsValidQuery());
+        },
+        [updateColumns]
+    );
+
+    const appendRow = useCallback(() => {
+        // Add row component
+        const rowId = String(Math.random());
+        const newRows = [
+            ...rows,
+            {
+                rowId: rowId
+            }
+        ];
+        updateRows(newRows);
+        // Set validity to false since row is just initialized
+        validityMap.current.set(rowId, false);
+        setIsSearchDisabled(!checkIsValidQuery());
+    }, [rows]);
+
+    const onSearch = useCallback(() => {
+        const query = buildQuery(Array.from(querySnippets.current.values()));
+        executeQuery(query);
+    }, [executeQuery]);
+
+    // Non-useCallback functions
+    const checkIsValidQuery = () => {
+        if (validityMap.current.size === 0) {
+            return true;
+        }
+        let isValid = true;
+        validityMap.current.forEach((validity: boolean) => {
+            if (!validity) {
+                isValid = false;
+            }
+        });
+        return isValid;
+    };
+
+    return (
+        <div className={classNames.root}>
+            <div className={classNames.headerGrid}>
+                {!(rows.length === 1) && (
+                    <p className={classNames.headerText}>
+                        {t('advancedSearch.andOr')}
+                    </p>
+                )}
+                <p className={classNames.headerText}>
+                    {t('advancedSearch.property')}
+                </p>
+                <p className={classNames.headerText}>
+                    {t('advancedSearch.operator')}
+                </p>
+                <p className={classNames.headerText}>
+                    {t('advancedSearch.value')}
+                </p>
+            </div>
+            <div className={classNames.rowContainer}>
+                {rows.map((row, index) => (
```

Comment:
Is it correct that the query does not include the parent property name for WheelInformation as is shown in the dropdown?
![image](https://user-images.githubusercontent.com/57726991/187563646-ba5732dd-46b6-41d7-b631-dbc8d899c3ba.png)